solutions/devsprint-leonardo-santos-2/FinanceApp/Components/ActivityCellView.swift

```diff
@@ -6,91 +6,160 @@
-
+//MARK: - ViewCodeProtocol
+extension ActivityCellView: ViewCodeProtocol {
+    
+    func configureHierarcy() {
         addSubview(mainStackView)
-        mainStackView.addArrangedSubview(categoryImageView)
-        mainStackView.addArrangedSubview(labelsStackView)
+        
+        [categoryImageView, labelsStackView]
+            .forEach(mainStackView.addArrangedSubview)
+        
+        [activityNameLabel, activityInfoLabel]
+            .forEach(labelsStackView.addArrangedSubview)
+    }
+    
+    func configureConstraint() {
+        
+        constrain(self, mainStackView) {
+            $1.safeAreaLayoutGuide.edges == $0.safeAreaLayoutGuide.edges
+        }
+        
+        constrain(categoryImageView) {
+            $0.height == Constant.imageSize
+            $0.width == Constant.imageSize
+        }
+    }
+    
+    func configureStyle() {
+        contentView.isHidden = true
+        accessoryView = accessoryViewAnimated
+    }
+}
 
-        labelsStackView.addArrangedSubview(activityNameLabel)
-        labelsStackView.addArrangedSubview(activityInfoLabel)
+//MARK: - SingleLopAnimationViewDelegate
+extension ActivityCellView: SingleLopAnimationViewDelegate {
+    func animationViewHandleTapped(
+        _ animationView: SingleLopAnimationView,
+        with animation: AnimationView
+    ) {
+        animationView.animate()
     }
+}
 
-    func configureConstraints() {
+private extension ActivityCellView {
+    struct Constant {
+        static let smallFontSize: CGFloat = 14
+        static let bigFontSize: CGFloat = 17
+        static let cornerRadius: CGFloat = 25
+        static let edgeLeftMargin: CGFloat = 16
+        static let mainStackSpacing: CGFloat = 8
+        static let imageSize: CGFloat = 50
+        static let accessorySize: CGFloat = 40
+        static let startFrame: TimeInterval = 0
+        static let endFrame: TimeInterval = 1
+        static let activityText = "$100.00 • 8:57 AM"
+        static let activityNameText = "Mall"
+    }
+}
 
-        NSLayoutConstraint.activate([
-            mainStackView.topAnchor.constraint(equalTo: safeAreaLayoutGuide.topAnchor),
-            mainStackView.bottomAnchor.constraint(equalTo: safeAreaLayoutGuide.bottomAnchor),
-            mainStackView.leadingAnchor.constraint(equalTo: safeAreaLayoutGuide.leadingAnchor),
-            mainStackView.trailingAnchor.constraint(equalTo: safeAreaLayoutGuide.trailingAnchor),
 
-            self.categoryImageView.widthAnchor.constraint(equalToConstant: 50),
-            self.categoryImageView.heightAnchor.constraint(equalToConstant: 50),
+protocol ViewCodeProtocol {
+    func configureViewCode()
+    func configureStyle()
+    func configureHierarcy()
+    func configureConstraint()
+}
 
-        ])
+extension ViewCodeProtocol {
+    func configureViewCode() {
+        configureStyle()
+        configureHierarcy()
+        configureConstraint()
     }
+    
+    func configureStyle() {  }
 }
 
+enum LottieFile: String {
```

Comment:
Se eu não me engano temos essa mesma classe em outro PR. Assim que ele for mergeado, poderia adicionar no mesmo enum?

Reply:
opa claro